File: src/main/java/com/fractalwrench/bpp/ByteCodeGenerator.java
package com.fractalwrench.bpp;

import com.fractalwrench.bpp.internal.ast.BppParser;
import org.parboiled.BaseParser;
import org.parboiled.Parboiled;

public class ByteCodeGenerator extends BaseParser {

    public byte[] generate(String sourceCode, String name) throws Exception {
        BppParser parser = Parboiled.createParser(BppParser.class);
        return parser.parse(sourceCode, name);
    }

}

File: src/main/java/com/fractalwrench/bpp/ByteCodeGenerator.kt
package com.fractalwrench.bpp

import com.fractalwrench.bpp.internal.ast.AstNode
import com.fractalwrench.bpp.internal.ast.BppParser
import org.parboiled.BaseParser
import org.parboiled.Parboiled

class ByteCodeGenerator : BaseParser<AstNode>() {

    @Throws(Exception::class)
    fun generate(sourceCode: String, name: String): ByteArray {
        val parser = Parboiled.createParser<BppParser, AstNode>(BppParser::class.java)
        return parser.parse(sourceCode, name)
    }

}
